features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/CheckboxSemanticsNodeMapper.kt

```diff
@@ -0,0 +1,205 @@
+    }
+
+    private fun createManualCheckedWireframe(
+        semanticsNode: SemanticsNode,
+        globalBounds: GlobalBounds,
+        backgroundColor: String
+    ): List<MobileSegment.Wireframe> {
+        val strokeColor = getFallbackCheckmarkColor(backgroundColor)
+
+        val background: MobileSegment.Wireframe = createUncheckedWireframes(
+            semanticsNode = semanticsNode,
+            globalBounds = globalBounds,
+            backgroundColor = backgroundColor
+        )[0]
+
+        val checkmarkWidth = globalBounds.width * CHECKMARK_SIZE_FACTOR
+        val checkmarkHeight = globalBounds.height * CHECKMARK_SIZE_FACTOR
+        val xPos = globalBounds.x + ((globalBounds.width / 2) - (checkmarkWidth / 2))
+        val yPos = globalBounds.y + ((globalBounds.height / 2) - (checkmarkHeight / 2))
+        val foreground: MobileSegment.Wireframe = MobileSegment.Wireframe.ShapeWireframe(
+            id = resolveId(semanticsNode, 1),
+            x = xPos.toLong(),
+            y = yPos.toLong(),
+            width = checkmarkWidth.toLong(),
+            height = checkmarkHeight.toLong(),
+            shapeStyle = MobileSegment.ShapeStyle(
+                backgroundColor = strokeColor,
+                opacity = 1f,
+                cornerRadius = CHECKBOX_CORNER_RADIUS
+            ),
+            border = MobileSegment.ShapeBorder(
+                color = DEFAULT_COLOR_BLACK,
+                width = BOX_BORDER_WIDTH_DP
+            )
+        )
+        return listOf(background, foreground)
+    }
+
+    private fun createUncheckedWireframes(
+        semanticsNode: SemanticsNode,
+        globalBounds: GlobalBounds,
+        backgroundColor: String
+    ): List<MobileSegment.Wireframe> {
+        val borderColor =
+            semanticsUtils.resolveBorderColor(semanticsNode)
+                ?.let { rawColor ->
+                    convertColor(rawColor)
+                } ?: DEFAULT_COLOR_BLACK
+
+        return listOf(
+            MobileSegment.Wireframe.ShapeWireframe(
+                id = resolveId(semanticsNode, 0),
+                x = globalBounds.x,
+                y = globalBounds.y,
+                width = globalBounds.width,
+                height = globalBounds.height,
+                shapeStyle = MobileSegment.ShapeStyle(
+                    backgroundColor = backgroundColor,
+                    opacity = 1f,
+                    cornerRadius = CHECKBOX_CORNER_RADIUS
+                ),
+                border = MobileSegment.ShapeBorder(
+                    color = borderColor,
+                    width = BOX_BORDER_WIDTH_DP
+                )
+            )
+        )
+    }
+
+    private fun isCheckboxChecked(semanticsNode: SemanticsNode): Boolean =
+        semanticsNode.config.getOrNull(SemanticsProperties.ToggleableState) == ToggleableState.On
+
+    private fun getFallbackCheckmarkColor(backgroundColor: String?) =
```

Comment:
it seems this function only work well when the background color is pure white or pure black? but is it designed to have the reversed color for any background color?

Reply:
It's in order to prevent having a white checkmark on a white background, which is the most common case. For all other cases it sets the checkmark to be black. Theoretically we could have a black checkmark on a black background so perhaps we could improve this

Reply:
This is probably needed for other mappers, let's create a ticket and handle it later